### PR TITLE
Avoid warnings about unreachable code

### PR DESCRIPTION
**Pull request by dkolsen-pgi**
NVC++ warns about unreachable code in specializations of `BaseDigitExtractor::ProcessFloatMinusZero` where `FLOAT_KEY` is false.  Rearrange the code to avoid the warning, without changing the behavior at all.

### CODE REVIEW

**Comment by dkolsen-pgi** [comment]
This is not urgent.  Wait until after the 1.13 release is out the door before merging it.  But it is simple enough that I would like it merged before too long.

**Comment by allisonvacanti** [comment]
It seems like the compiler shouldn't warn on this, since the condition is parametrized and either branch could be valid. Why not just discard the dead code silently?

LGTM otherwise. Since it's a harmless change I'll go ahead and test / merge this once gpuCI is working again, but this is surprising behavior.

**Comment by dkolsen-pgi** [comment]
You are correct that the compiler shouldn't warn about this.  Before creating this PR, I filed a bug against EDG (our compiler front end) to get rid of the warning.  It will take a while for the compiler to get fixed, so I would like to change the CUB source to avoid the warning in the meantime.

**Comment by allisonvacanti** [comment]
Gotcha -- sounds good to me.

Rebased to pull in recent CI fixes.

DVS CL: 30167030
gpuCI: NVIDIA/thrust#1482

**Comment by allisonvacanti** [comment]
Rebased for more CI fixes...

DVS CL: 30211051
gpuCI: NVIDIA/thrust#1482

**Review by allisonvacanti — APPROVED** [review]
Tests pass, LGTM!

**Pull request merged**
